src/serde.rs

```diff
@@ -1,4 +1,12 @@
-//! Serialization/Deserialization traits and concrete implementations for handlers and state.
+//! # Serialization
+//!
+//! Restate sends data over the network for storing state, journaling actions, awakeables, etc.
+//!
+//! Therefore, all data structures need to have serializers and deserializers.
+//!
+//! You can use [Rust Serde](https://serde.rs/) for this.
+//! Either [use derive](https://serde.rs/derive.html) or implement the [`Serialize` and `Deserialize` traits](https://serde.rs/custom-serialization.html).
```

Comment:
these couple of sentences are very weirdly structured, and i don't think they're correct.

The story here is that we have our own `Serialize` and `Deserialize` traits (which are NOT the serde.rs traits!), and those have already implementations for primitive types. If you want to serialize structs/enums as json, you can use the `Json` wrapper below which is using serde.rs

Reply:
hmm okay, but this enum works for storing in state:
```
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub(crate) enum TicketStatus {
    Available,
    Reserved,
    Sold,
}
```

Would you mind just rewriting the section as a suggestion so I can just commit it. I struggle to find the right wording.

Reply:
what you sent works If you wrap it in `Json`